Objects/frameobject.c

```diff
@@ -613,7 +613,7 @@ frame_dealloc(PyFrameObject *f)
 
     /* Kill all local variables */
     if (f->f_localsptr) {
-        for (int i = 0; i < co->co_nlocalsplus+FRAME_SPECIALS_SIZE; i++) {
+        for (int i = 0; i < co->co_nfastlocals+FRAME_SPECIALS_SIZE; i++) {
```

Comment:
Either this is wrong, or `co_nfastlocals` is misnamed.
This should clear the fastlocals, cell and free variables (plus the specials), not just the "fast" locals.